build_defs/java.build_defs

```diff
@@ -539,6 +539,17 @@ def _java_toolchain(name : str, jdk_url : str|dict='', jdk:str='', visibility:li
             hashes = hashes,
         )
 
+    if jdk:
+        if CONFIG.OS == "linux" or CONFIG.OS == "darwin" or CONFIG.OS == "freebsd" or CONFIG.OS == "openbsd":
+            java_platform = CONFIG.OS
+        elif CONFIG.OS == "windows":
+            java_platform = "win32"
```

Comment:
I wonder if this would be better:
```
if CONFIG.OS == "windows":
    java_platform = "win32"
else:
    java_platform = CONFIG.OS
```

Unless you're certain these are the set of valid OSs?

Reply:
This is the [list of Java platform names that Bazel recognises](https://bazel.googlesource.com/bazel/+/refs/tags/3.4.0/tools/jdk/BUILD#147), so we should be safe to go with it. I think there's also the possibility of `include/solaris` for Solaris, which I could add (do we have any Please users running Solaris?).

Reply:
I was thinking more about having less hard coded stuff in the build def, rather than to actually support other platforms. It doesn't really matter though.